streaming/base/stream.py

```diff
@@ -94,11 +96,11 @@ def __init__(self,
                  choose: Optional[int] = None,
                  download_retry: Optional[int] = None,
                  download_timeout: Optional[float] = None,
+                 local_directory_timeout: float = 60,
```

Comment:
I would recommend not to add another parameter. Can we use `download_timeout` instead ? I don't think `mkdirs()` can take more than 1 sec, so we don't need to worry much about timeout but it's a good habit to raise an exception which you did.